### DISCUSSION TITLE
Deployment with Traefik

### DISCUSSION THREAD

**Discussion by marcosmmb** [fastapi/fastapi#8141] [Questions] [answered]
I managed to configure Traefik with Docker but how can I make my FastAPI application point to it? I already can access Traefik and Consul UI, but I'm unable to access my application on port 8000 with SSL.

**Comment by wshayes** [comment, marked as answer]
You should terminate SSL at Traefik and then traefik will forward the request to your fastapi container as http.  You can add something like the following (see the labels section):

    networkstore:
      container_name: pynetworkstore
      image: biodati/pynetworkstore:localdev
      build:
        context: ./pynetworkstore
        dockerfile: ./docker/Dockerfile
      environment:
        - LOGGING=INFO
      depends_on:
        - arangodb
        - elasticsearch
      command: ["/start-reload.sh"]  # Used for development only
      volumes:
        - ./pynetworkstore/app:/app
      networks:
        biodati:
      labels:
        - traefik.enable=true
        - traefik.backend=networkstore
        - traefik.frontend.rule=Host:networkstore.${BS_HOST_NAME:?err};
        - traefik.port=80

The traefik labels tell traefik to route traffic for networkstore.example.test:80 to the networkstore container.

**Comment by wshayes** [comment]
These types of questions are a better fit for the Gitter FastAPI channel (https://gitter.im/tiangolo/fastapi) or the Traefik Slack channel.

**Comment by tiangolo** [comment]
Thanks @wshayes for the help!

@marcosmmb you can check the project generators, they have everything already prepared to be connected to Traefik: https://fastapi.tiangolo.com/project-generation/

**Comment by marcosmmb** [comment]
@tiangolo I found a more interesting solution for me. I'm using Caprover, which works as a self hosted Heroku, so I don't need to worry about all that stuff. Since Caprover works with docker images, it is pretty easy to launch a FastAPI container with it. If you like, I could PR a tutorial for that.

**Comment by tiangolo** [comment]
Great @marcosmmb ! Thanks for reporting back about your solution. I think that would make an interesting blog post :wink: 

As I see you already solved your problem, I'm going to close this issue now. But feel free to add more comments or issues.
